pkg/worker/job.go

```diff
@@ -77,9 +80,14 @@ func contextLogger(ctx context.Context, job *Job) context.Context {
 		return ctx
 	}
 
+	// if otel enabled
+	var span trace.Span
+	ctx, span = otel.Tracer(telemetry.TracePrefix+"pkg/worker").Start(ctx, string(job.Type))
+
 	accountId := job.AccountID
 	id := job.Identity
 	logger := zerolog.Ctx(ctx).With().
+		Str("trace_id", span.SpanContext().TraceID().String()).
 		Str("job_id", job.ID.String()).
 		Int64("account_id", accountId).
 		Str("account_number", id.Identity.AccountNumber).
```

Comment:
Yeah this makes a lot of sense. In my patch, I put trace_id at reservation level but I think this is much better place. I will move it here but I will not add any otel related stuff and leave this to you so you can further develop this.

Great idea, in the end, we will have one trace (aka transaction in GlitchTip) and spans for api pod then one big span for the worker pod and additional spans for individual functions.